test/client/fixtures/transport-test.js

```diff
@@ -254,6 +254,79 @@ else {
             });
     });
 }
+if (browserUtils.isWebKit) {
+    asyncTest("do not resend aborted async service msg if it contains 'disableResending' flag (WebKit)", function () {
+        settings.get().sessionId = '%%%testUid%%%';
+
+        var xhrCount      = 0;
+        var isMessageSent = false;
+
+        ok(!window.localStorage.getItem(settings.get().sessionId));
+
+        var onAjaxSend = function (xhr) {
+            xhrCount++;
+            xhr.abort();
+        };
+
+        reqisterAfterAjaxSendHook(onAjaxSend);
+
+        var msg = {
+            disableResending: true
+        };
+
+        transport.asyncServiceMsg(msg)
+            .then(function () {
+                isMessageSent = true;
+            });
+
+        setTimeout(function () {
+            strictEqual(xhrCount, 1);
+
+            var storedMsgStr   = window.localStorage.getItem(settings.get().sessionId);
+            var storedMsgCount = JSON.parse(storedMsgStr).length;
+
+            notOk(isMessageSent);
+            strictEqual(storedMsgStr, '[]');
+            strictEqual(storedMsgCount, 0);
```

Comment:
you check the `storedMsgStr`, so the `storedMsgCount` check is redundant. It can be failed only if the previous check is failed too